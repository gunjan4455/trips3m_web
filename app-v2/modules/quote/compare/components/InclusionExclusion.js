import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { BlueDownArrow } from 'app/helpers/Icon/Icon';
import CompareList from './CompareList';
import CompareTextSection from './CompareTextSection';
import { GCompareCardContentDiv, GCardViewActionDiv } from '../G';
import { ORGANISM_CLICKED } from 'actions/segmentEvents';

class InclusionExclusion extends Component {
  static propTypes = {
    quotes: PropTypes.object.isRequired,
    trackSegment: PropTypes.func.isRequired,
    isExclusion: PropTypes.bool
  };

  static defaultProps = {
    isExclusion: false
  };

  state = {
    showMore: false,
    text: 'View More'
  };

  showAndHideSection = () => {
    if (this.state.showMore) {
      this.setState({ showMore: !this.state.showMore, text: 'View More' });
    } else {
      this.setState({ showMore: !this.state.showMore, text: 'View Less' });
    }

    this.props.trackSegment({
      event: ORGANISM_CLICKED,
      section: this.props.isExclusion ? 'Exclusion' : 'Inclusion',
      cta: this.state.showMore ? 'View More' : 'View Less',
      miscId1: Object.keys(this.props.quotes).join('::'),
      miscType1: 'quote_id_left::quote_id_right'
    });
  };

  render() {
    const { isExclusion, quotes } = this.props;
    const { showMore } = this.state;
    let isViewMore = false;

    const compareItems = Object.keys(quotes).map((id) => {
      const quote = quotes[id];
      let inclusions = isExclusion ? quote.inclusions['0'] : quote.inclusions['1'];
      if (inclusions === undefined) {
        return (<CompareTextSection
          key={`inc_${id}`}
          text={isExclusion ? quote.exclusionsText : quote.inclusionsText}
        />);
      }

      if (inclusions.length > 5) {
        isViewMore = true;
        if (!showMore) {
          inclusions = inclusions.slice(0, 5);
        }
      }

      return (
        <CompareList
          key={`inc_${id}`}
          inclusions={inclusions}
          isExclusion={isExclusion}
        />
      );
    });

    return (
      <div>
        <GCompareCardContentDiv className="flex at_compareCardContent">
          {compareItems}
        </GCompareCardContentDiv>
        {
          isViewMore
            ? <GCardViewActionDiv className="clearfix">
              <button
                className={`f14 fw7 sfc1 mr15 mb15 fright flex alignCenter at_qcp_viewMore ${showMore ? 'cardViewLess' : ''}`}
                onClick={this.showAndHideSection}>
                {this.state.text}
                <span className="relative iblock icon-14 ml8">
                    <BlueDownArrow />
                  </span>
              </button>
            </GCardViewActionDiv>
            : null
        }
      </div>
    );
  };
}

export default InclusionExclusion;
