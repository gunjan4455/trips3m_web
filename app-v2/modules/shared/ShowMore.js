import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import _isEmpty from 'lodash/isEmpty';

import { BlueDownArrow } from 'helpers/Icon/Icon';
import { ORGANISM_CLICKED } from 'actions/segmentEvents';

const GreadmoreLabel = styled.label({
  '& button.read-more-button': {
    lineHeight: '20px',
    padding: '0',
  },
});

const GDownArrowSpan = styled.span({
  width: '10px',
  height: '6px',
  '& svg': {
    width: '10px',
    height: '6px',
    top: '1px'
  }
});

class ShowMore extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      checked: props.defaultOpen
    };
    this.handleChange = this.handleChange.bind(this);
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({
      checked: false
    });
  }

  handleChange(e) {
    e.preventDefault();

    this.setState({
      checked: !this.state.checked
    }, () => {
      this.state.checked && this.props.trackSegment({ cta: 'Read More' });
    });
  }

  render() {
    const textLength = 195;
    const { alignCenter, noOverwrite } = this.props;
    return (
      <div className="row row- text-left">
        <input type="checkbox" className="read-more-state" checked={this.state.checked} id="post-1" />
        <div
          className={`read-more-wrap ${this.props.isDefaultShowMore ? 'p-last-no-margin pfc3 f14p dynamicTextInherit' : ''} ${noOverwrite ? 'noOverwrite' : ''}`}>
          {
            // TODO: find a better way for truncate the string
            !this.state.checked && this.props.appendText && this.props.wrapText.length >= textLength
              ? this.props.wrapText + this.props.appendText
              : this.props.wrapText
          }
          <span className="read-more-target">
            {this.props.moreText}
          </span>
        </div>
        {
          this.props.moreText.length > 0 || !_isEmpty(this.props.moreText) ?
            <GreadmoreLabel htmlFor="post-1" className={`${alignCenter ? 'flex justifyCenter p15 bt row' :'read-more-trigger'} relative`}>
              <button
                className="read-more-button flex alignCenter spaceBetween relative f12 fw7"
                onClick={this.handleChange}>
                <span className="flexFull f14 fw7">
                  {(this.state.checked) ? this.props.showLessText : this.props.showMoreText}
                  {this.state.checked}
                </span>
                <GDownArrowSpan
                  className="flex ml8 relative alignCenter justifyCenter"><BlueDownArrow /></GDownArrowSpan>
              </button>
            </GreadmoreLabel>
            : ''
        }
      </div>
    );
  }
}

ShowMore.propTypes = {
  wrapText: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.string,
    PropTypes.object
  ]).isRequired,
  moreText: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.string,
    PropTypes.object
  ]),
  showMoreText: PropTypes.string,
  showLessText: PropTypes.string,
  alignCenter: PropTypes.bool,
  defaultOpen: PropTypes.bool,
  isDefaultShowMore: PropTypes.bool,
  trackSegment: PropTypes.func,
  appendText: PropTypes.string,
  noOverwrite: PropTypes.bool,
};

ShowMore.defaultProps = {
  wrapText: '',
  moreText: '',
  showMoreText: 'Read More',
  showLessText: 'Read Less',
  defaultOpen: false,
  isDefaultShowMore: true,
  alignCenter: false,
  trackSegment: () => {
  },
  noOverwrite: false
};

export default ShowMore;
