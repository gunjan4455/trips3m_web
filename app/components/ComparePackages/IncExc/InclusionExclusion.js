import React from 'react';
import PropTypes from 'prop-types';
import InclusionExclusionRow from './InclulsionExclusionRow';
import '../ComparePackages.scss';



/* eslint-disable react/no-array-index-key */
const renderInclusionExclusionRows = (inclusions, uniqueInclusions) =>
  uniqueInclusions.map((uniqueInclusion, i) =>
    <InclusionExclusionRow key={i} inclusions={inclusions} uniqueInclusion={uniqueInclusion} />);
/* eslint-enable */

const InclusionExclusion = ({ inclusions, uniqueInclusions }) => (
  <div className='col-xs-12 p0 sbcw accordian-contenido'>
    { renderInclusionExclusionRows(inclusions, uniqueInclusions) }
  </div>
);

InclusionExclusion.propTypes = {
  inclusions: PropTypes.array.isRequired,
  uniqueInclusions: PropTypes.array.isRequired
};

export default InclusionExclusion;
