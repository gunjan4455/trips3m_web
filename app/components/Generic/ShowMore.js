import React from 'react';
import PropTypes from 'prop-types';
import {
  BlueDownArrow
} from 'helpers/Icon/Icon';
import { PACKAGE_INTERACTED } from "../../actions/segmentEvents";

class ShowMore extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      checked: false
    };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange() {
    this.setState({
      checked: !this.state.checked
    });
    this.props.trackSegment(PACKAGE_INTERACTED, 'Overview', (this.state.checked ?
      this.props.showLessText : this.props.showMoreText ));
  }

  render() {
    const { isRatingSchemaEnabled, showLessText, showMoreText } = this.props;
    return (
      <div>
        <input type="checkbox" className="read-more-state" checked={this.state.checked} id="post-1"/>
        <div className="read-more-wrap" property={isRatingSchemaEnabled ? 'description' : null}>{this.props.wrapText}<span className="read-more-target">{this.props.moreText}</span></div>
        <label htmlFor="post-1" className="read-more-trigger">
          <button className="read-more-button" onClick={this.handleChange}>
            {(this.state.checked) ? showLessText : showMoreText}
            <BlueDownArrow />
          </button>
        </label>
      </div>
    );
  }
}
ShowMore.propTypes = {
  wrapText: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
    PropTypes.string,
  ]).isRequired,
  moreText: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
    PropTypes.string,
  ]),
  showMoreText: PropTypes.string,
  showLessText: PropTypes.string,
  trackSegment: PropTypes.func,
  isRatingSchemaEnabled: PropTypes.bool
};

ShowMore.defaultProps = {
  moreText: '',
  showMoreText: 'Read More',
  showLessText: 'Read Less',
  trackSegment: () => {},
  isRatingSchemaEnabled: false
};

export default ShowMore;
