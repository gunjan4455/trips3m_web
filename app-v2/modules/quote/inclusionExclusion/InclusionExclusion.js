import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { Element } from 'react-scroll';

import { PDM_CATEGORY_INCLUSION } from 'helpers/pdmHelpers';
import { ORGANISM_CLICKED } from 'actions/segmentEvents';
import ShowMore from 'modules/shared/ShowMore';
import QuoteCardHeader from "modules/quote/shared/QuoteCardHeader";
import {InclusionsNewIcon, ExclusionsNewIcon} from "helpers/Icon/Icon";

const GListItemDiv = styled.div(({ title }) => ({
  paddingLeft: '25px',
  position: 'relative',
  '&:after': {
    content: '\' \'',
    width: '8px',
    height: '8px',
    background: title === 'Exclusions' ? '#fc5145' : '#4caf50',
    borderRadius: '100%',
    position: 'absolute',
    left: '0',
    top: '6px'
  }
}));

function hideTitle(item) {
  return item.category === 'Others' && (item.comments === null ||
    item.comments === '');
}

const renderComments = (comments) => (comments ? <span> : {comments}</span> : null);

const renderListItem = (data, title) => (
  data.map((listItem, index) => hideTitle(listItem)
    ? ''
    : <GListItemDiv key={listItem.id || index} className="mb15" title={title}>
      {listItem.name || listItem}
      {renderComments(listItem.comments)}
    </GListItemDiv>)
);

const renderList = (listItems, classNames, title, trackSegmentEvent) => {
  if (listItems.length > 4) {
    const wrapData = listItems.slice(0, 4);
    const moreData = listItems.slice(5);

    const wrapText = renderListItem(wrapData, title);
    const moreText = renderListItem(moreData, title);

    return (<div className={"sbcw p15 pt0 pl24" + classNames}><ShowMore
      wrapText={wrapText}
      moreText={moreText}
      showMoreText="Read More"
      showLessText="Read Less"
      trackSegment={trackSegmentEvent}
    /></div>);
  }

  return (<div className={"sbcw mr8 ml8 p15" + classNames}>
    {renderListItem(listItems, title)}
  </div>);
};

const isInclusion = (sectionName) => sectionName === 'inclusions-sec';

const InclusionExclusion = ({
                              title, item, classNames, sectionName,
                              setPdmCategory, showHidePdmView, trackSegment
                            }) => {

  const trackSegmentEvent = ({ cta }) => {
    trackSegment({
      event: ORGANISM_CLICKED,
      section: sectionName,
      cta: cta
    });
  };

  const handleChatClick = () => {
    if (isInclusion(sectionName)) {
      setPdmCategory(PDM_CATEGORY_INCLUSION);
      showHidePdmView(true);
    }

    trackSegmentEvent({ cta: 'Chat' });
  };

  return (
    item ?
      <Element name={sectionName}>
        <div className="pfc3 f14 sbcw mt8">
          <QuoteCardHeader heading={title}
                           chatButtonName={isInclusion(sectionName) ? 'Edit Package' : 'Chat'}
                           handleChatClick={handleChatClick}
          icon={isInclusion(sectionName) ? <InclusionsNewIcon /> : <ExclusionsNewIcon />}/>
          {
            typeof item === 'string'
              ? <div className="sbcw p15 pl48 pt0">{item}</div>
              : renderList(item, classNames, title, trackSegmentEvent)
          }
        </div>
      </Element> : null
  );
};

InclusionExclusion.propTypes = {
  title: PropTypes.string,
  item: PropTypes.oneOfType([PropTypes.string, PropTypes.array]).isRequired,
  classNames: PropTypes.string,
  sectionName: PropTypes.string,
  setPdmCategory: PropTypes.func,
  showHidePdmView: PropTypes.func,
  trackSegment: PropTypes.func
};

InclusionExclusion.defaultProps = {
  title: '',
  classNames: '',
  sectionName: '',
  setPdmCategory: () => {},
  showHidePdmView: () => {},
  trackSegment: () => {}
};

export default InclusionExclusion;
