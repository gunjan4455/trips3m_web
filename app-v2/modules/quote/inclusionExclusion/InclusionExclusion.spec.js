import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow, configure } from 'enzyme';

import InclusionExclusion from './InclusionExclusion';
import ShowMore from 'modules/shared/ShowMore';

configure({ adapter: new Adapter() });
let component;

const DEFAULT_PROPS = {
  title: 'Inclusions',
  item: 'Inclusion dummy text',
  classNames: '',
  sectionName: 'inclusions-sec',
};

const shallowWrappperInclusionComponent = (props) => {
  return shallow(<InclusionExclusion {...props} />);
};

describe('Inclusions: Render Correctly ', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should match with the previous snapshot', () => {
    const wrapper = shallowWrappperInclusionComponent(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with inclusions array', () => {
    props.item = [
      {
        id: 272,
        name: 'Double',
        category: 'Accomodation',
        comments: null
      },
      {
        id: 274,
        name: 'Breakfast',
        category: 'Meal plan',
        comments: null
      },
      {
        id: 276,
        name: 'Dinner',
        category: 'Meal plan',
        comments: null
      },
      {
        id: 279,
        name: 'Welcome Drink',
        category: '',
        comments: null
      },
      {
        id: 281,
        name: 'Wifi Connection',
        category: '',
        comments: null
      },
      {
        id: 284,
        name: 'Transfers (Railway/Airport)',
        category: '',
        comments: null
      },
      {
        id: 286,
        name: 'Tv with cable connection',
        category: '',
        comments: null
      },
      {
        id: 289,
        name: 'English Speaking Driver',
        category: '',
        comments: null
      },
      {
        id: 277,
        name: 'Arrival',
        category: 'Transfers (Railway/Airport)',
        comments: ''
      },
      {
        id: 278,
        name: 'Departure',
        category: 'Transfers (Railway/Airport)',
        comments: ''
      }
    ];

    const wrapper = shallowWrappperInclusionComponent(props);
    expect(wrapper.find(ShowMore)).toHaveLength(1);
  });
});
