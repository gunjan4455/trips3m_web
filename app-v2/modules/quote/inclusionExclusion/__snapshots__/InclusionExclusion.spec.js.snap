// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inclusions: Render Correctly  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InclusionExclusion
    classNames=""
    item="Inclusion dummy text"
    sectionName="inclusions-sec"
    setPdmCategory={[Function]}
    showHidePdmView={[Function]}
    title="Inclusions"
    trackSegment={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="pfc3 f14 sbcw mt8"
      >
        <QuoteCardHeader
          category=""
          chatButtonName="Edit Package"
          handleChatClick={[Function]}
          heading="Inclusions"
          icon={<IconSvg />}
        />
        <div
          className="sbcw p15 pl48 pt0"
        >
          Inclusion dummy text
        </div>
      </div>,
      "name": "inclusions-sec",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <QuoteCardHeader
            category=""
            chatButtonName="Edit Package"
            handleChatClick={[Function]}
            heading="Inclusions"
            icon={<IconSvg />}
          />,
          <div
            className="sbcw p15 pl48 pt0"
          >
            Inclusion dummy text
          </div>,
        ],
        "className": "pfc3 f14 sbcw mt8",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "category": "",
            "chatButtonName": "Edit Package",
            "handleChatClick": [Function],
            "heading": "Inclusions",
            "icon": <IconSvg />,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Inclusion dummy text",
            "className": "sbcw p15 pl48 pt0",
          },
          "ref": null,
          "rendered": "Inclusion dummy text",
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="pfc3 f14 sbcw mt8"
        >
          <QuoteCardHeader
            category=""
            chatButtonName="Edit Package"
            handleChatClick={[Function]}
            heading="Inclusions"
            icon={<IconSvg />}
          />
          <div
            className="sbcw p15 pl48 pt0"
          >
            Inclusion dummy text
          </div>
        </div>,
        "name": "inclusions-sec",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <QuoteCardHeader
              category=""
              chatButtonName="Edit Package"
              handleChatClick={[Function]}
              heading="Inclusions"
              icon={<IconSvg />}
            />,
            <div
              className="sbcw p15 pl48 pt0"
            >
              Inclusion dummy text
            </div>,
          ],
          "className": "pfc3 f14 sbcw mt8",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "category": "",
              "chatButtonName": "Edit Package",
              "handleChatClick": [Function],
              "heading": "Inclusions",
              "icon": <IconSvg />,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Inclusion dummy text",
              "className": "sbcw p15 pl48 pt0",
            },
            "ref": null,
            "rendered": "Inclusion dummy text",
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
